src/complex/Utilities/StringUtilities.hpp

```diff
@@ -0,0 +1,224 @@
+ * OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE
+ * USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+ *
+ *
+ * ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~ */
+#pragma once
+
+#include <array>
+#include <cctype>
+#include <regex>
+#include <sstream>
+#include <string>
+#include <vector>
+
+/*' '(0x20)space(SPC)
+ * '\t'(0x09)horizontal tab(TAB)
+ * '\n'(0x0a)newline(LF)
+ * '\v'(0x0b)vertical tab(VT)
+ * '\f'(0x0c)feed(FF)
+ * '\r'(0x0d)carriage return (CR)
+ */
+
+namespace complex
+{
+namespace StringUtilities
+{
+
+using StringTokenType = std::vector<std::string>;
+
+// 5 statements
+template <class InputIt, class ForwardIt, class BinOp>
+void for_each_token(InputIt first, InputIt last, ForwardIt s_first, ForwardIt s_last, BinOp binary_op)
+{
+  while(true)
+  {
+    const auto pos = std::find_first_of(first, last, s_first, s_last);
+    binary_op(first, pos);
+    if(pos == last)
+    {
+      break;
+    }
+    first = std::next(pos);
+  }
+}
+
+// 2 statements
+inline StringTokenType split(const std::string& str, char delim)
+{
+  StringTokenType tokens;
+  std::string temp(str);
+  std::array<char, 1> delims = {delim};
+  auto endPos = std::end(temp);
+  for_each_token(std::begin(temp), endPos, std::cbegin(delims), std::cend(delims), [&endPos, &tokens](auto first, auto second) {
+    if(first != second)
+    {
+      if(second != endPos)
+      {
+        *second = '\0';
+      }
+      tokens.push_back({&*first});
+    }
+  });
+  // std::cout << "Tokens: " << tokens.size() << std::endl;
+  return tokens;
+}
+
+inline StringTokenType split_2(const std::string& line, char delimiter)
+{
+  std::stringstream ss(line);
+
+  StringTokenType tokens;
+  std::string temp_str;
+
+  while(getline(ss, temp_str, delimiter))
+  {
+    tokens.push_back(temp_str);
+  }
+  return tokens;
+}
+
+inline std::string replace(std::string str, const std::string& from, const std::string& to)
+{
+  size_t start_pos = 0;
+  while((start_pos = str.find(from, start_pos)) != std::string::npos)
+  {
+    str.replace(start_pos, from.length(), to);
+    start_pos += to.length();
+  }
+  return str;
+}
+
+inline std::string ltrim(const std::string& s)
+{
+  std::string out = s;
+  if(out.empty())
+  {
+    return out;
+  }
+  std::string whitespaces(" \t\f\v\n\r");
+  std::string::size_type front = out.find_first_not_of(whitespaces);
+  if(front != std::string::npos)
+  {
+    out = out.substr(front);
+  }
+  else
+  {
+    out.clear();
+  }
+  return out;
+}
+
+inline std::string rtrim(const std::string& s)
+{
+  std::string out = s;
+  if(out.empty())
+  {
+    return out;
+  }
+  std::string whitespaces(" \t\f\v\n\r");
+  std::string::size_type back = out.find_last_not_of(whitespaces);
+  if(back != std::string::npos)
+  {
+    out.erase(back + 1);
+  }
+  else
+  {
+    out.clear();
+  }
+  return out;
+}
+
+inline std::string trimmed(const std::string& s)
```

Comment:
Is there a reason this does not just call `ltrim` and `rtrim` and be done with it? With these being inlined, it shouldn't impose a penalty when compiling, and it's a lot easier to read that way than figuring out how it's different or keeping everything updated should changes be made.